acoustid/future/fingerprint.py

```
@@ -0,0 +1,50 @@
+        data: Compressed bytes containing the fingerprint data.
+
+    Returns:
+        Tuple containing a list of integer hash values and the version of the fingerprint.
+
+    The function decompresses the zstd data, unpacks the bytes into integers,
+    and reconstructs the original hash values by reversing the XOR differences.
+    """
+    data = zstd.decompress(data)
+    magic, format, version, *diffs = struct.unpack(f"<HBB{len(data) // 4 - 1}i", data)
+    if magic != MAGIC:
+        raise ValueError("Invalid fingerprint magic")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)